### DISCUSSION TITLE
How to react/persist UI state when the user quits the main appplication?

### DISCUSSION THREAD

**Discussion by alexeyabel** [slint-ui/slint#6392] [Q&A] [answered]
Can I somehow react (in Rust) to the user closing the main window, i.e. clicking the X of the window manager or pressing ALT+F4? I want to persist the UI state.

The alternative AFAICS would be to persist after each change, but the former seems to be more "efficient", I am not sure though. I would appreciate an opinion on what the best-practice is for this scenario.

**Comment by Vadoola** [comment, marked as answer]
I believe this is what you are looking for: https://releases.slint.dev/1.8.0/docs/rust/slint/struct.window#method.on_close_requested

**Reply by alexeyabel** [reply]
Thank you, I had searched for "quit" and "exit", but not "close".
